### ISSUE TITLE
Deduplicate i18n code and make use of localizedString widget

### ISSUE THREAD

**Issue by Bakkhos** [hardforkio/homepage#309]
Followup to #281

Support for localizedString widget was added in #282. This also refactored the entire i18n system, for instance it changed the implementation of the `TranslationCollection<T>` to
```typescript
type TranslationCollection<T> = ReadonlyArray<LanguageVersion<T>>
```
from
```typescript
interface TranslationCollection<T> {
  translations: LanguageVersion<T>[]
}
```
The new version of the system lives in [/src/cms/i18n.ts](https://github.com/hardforkio/homepage/blob/5a8fa1e828e1f6daa6cc290acdb34ffd3a982476/src/cms/i18n.ts) and its dependencies; but it's only used for the footer collection, while all other collections still use [/src/data/i18n.tsx](https://github.com/hardforkio/homepage/blob/5a8fa1e828e1f6daa6cc290acdb34ffd3a982476/src/data/i18n.tsx)

To avoid having two parallel systems and to see if the new design works, refactor everything to use the new system: refactor all schemas to use the new TranslationCollection and/or the localizedString widget; migrate the json data; replace the implementations in src/data/i18n.tsx with the new implementations (e.g. replace [filterByLocale](https://github.com/hardforkio/homepage-new/blob/5a8fa1e828e1f6daa6cc290acdb34ffd3a982476/src/data/i18n.tsx#L26-L27) with [translate](https://github.com/hardforkio/homepage-new/blob/5a8fa1e828e1f6daa6cc290acdb34ffd3a982476/src/cms/i18n.ts#L6-L7)), finally update types like this one
https://github.com/hardforkio/homepage/blob/5a8fa1e828e1f6daa6cc290acdb34ffd3a982476/src/data/home/index.tsx#L26-L39)

**Comment by Bakkhos** [comment]
As discussed: If you find out that it was a bad idea to change the `TranslationCollection<T>`, undo that change and bring the footer back to the old system

**Comment by EmCeeEs** [comment]
Fixed by #341
